packages/test/test-end-to-end-tests/src/test/blobsisAttached.spec.ts

```diff
@@ -0,0 +1,201 @@
+			assert.strictEqual(blob.isAttached, false);
+			directory.set(testKey, blob);
+			assert.strictEqual(blob.isAttached, true);
+		});
+	});
+
+	describe("from detached container", () => {
+		let provider: ITestObjectProvider;
+		let loader: IHostLoader;
+		let container: IContainer;
+		let detachedBlobStorage: MockDetachedBlobStorage;
+		let detachedDataStore: ITestFluidObject;
+		let map: SharedMap;
+		let directory: SharedDirectory;
+		let text: string;
+		let blobHandle: IFluidHandle<ArrayBufferLike>;
+
+		beforeEach(async () => {
+			provider = getTestObjectProvider();
+			detachedBlobStorage = new MockDetachedBlobStorage();
+			loader = provider.makeTestLoader({
+				...testContainerConfig,
+				loaderProps: { detachedBlobStorage },
+			});
+			container = await loader.createDetachedContainer(provider.defaultCodeDetails);
+			provider.updateDocumentId(container.resolvedUrl);
+			detachedDataStore = await requestFluidObject<ITestFluidObject>(container, "default");
+			map = SharedMap.create(detachedDataStore.runtime);
+			directory = SharedDirectory.create(detachedDataStore.runtime);
+			text = "this is some example text";
+			blobHandle = await detachedDataStore.runtime.uploadBlob(stringToBuffer(text, "utf-8"));
+		});
+
+		const checkForDetachedHandles = (dds: SharedMap | SharedDirectory) => {
+			assert.strictEqual(
+				container.attachState,
+				AttachState.Detached,
+				"container should be detached",
+			);
+			assert.strictEqual(
+				detachedDataStore.handle.isAttached,
+				false,
+				"data store handle should be detached",
+			);
+			assert.strictEqual(dds.handle.isAttached, false, "dds handle should be detached");
+			assert.strictEqual(blobHandle.isAttached, false, "blob handle should be detached");
+		};
+
+		it("all detached", async function () {
+			checkForDetachedHandles(map);
+			checkForDetachedHandles(directory);
+		});
+
+		it("after map is set in root directory", async function () {
+			detachedDataStore.root.set(mapId, map.handle);
+			checkForDetachedHandles(map);
+		});
+
+		it("after directory is set in root directory", async function () {
+			detachedDataStore.root.set(directoryId, directory.handle);
+			checkForDetachedHandles(directory);
+		});
+
+		it("after blob handle is set in map", async function () {
+			detachedDataStore.root.set("map", map.handle);
+			map.set("my blob", blobHandle);
+			checkForDetachedHandles(map);
+		});
+
+		it("after blob handle is set in directory", async function () {
+			detachedDataStore.root.set(directoryId, directory.handle);
+			directory.set("my blob", blobHandle);
+			checkForDetachedHandles(directory);
+		});
+
+		itExpects(
+			"after container is attached with map",
+			ContainerCloseUsageError,
+			async function () {
+				detachedDataStore.root.set("map", map.handle);
+				map.set("my blob", blobHandle);
+				const attachP = container.attach(
+					provider.driver.createCreateNewRequest(provider.documentId),
+				);
+				if (!driverSupportsBlobs(provider.driver)) {
```

Comment:
i think i would rather have the test skip itself rather than throw and handle an error.